# Configuration parameters for EGM/RWS connections to the YuMi Robot
rws:
  ip: 192.168.122.122
  port: 8080
  delay_time: 1.0
  max_signal_retries: 5

# EGM parameters, documentation taken from the RAPID module TRobEGM.sys
# TODO add the other parameters that are in egm settings as well here. 
egm:
  port_l: 6511
  port_r: 6512
  comm_timeout: 300.0
  
  # tool and work object names,
  # not important for YuMi since EGM control is at the joint level
  # tool_name: tool0
  # wobj_name: wobj0

  # Condition value [deg or mm] for when the EGM correction is considered
  # to be finished. E.g. for joint mode, then the condition is fulfilled when
  # the joints are within [-MinMax, MinMax].Condition value [deg or mm] 
  # for when the EGM correction is considered to be finished. E.g. for joint mode
  # then the condition is fulfilled when the joints are within [-MinMax, MinMax].
  cond_min_max: 0.0
  
  # Low pass filter bandwidth of the EGM controller [Hz].
  lp_filter: 1.0

  # Maximum admitted joint speed change [deg/s]:
  # Note: Should not be set higher than the lowest max speed [deg/s],
  # out of all the axis max speeds (found in the robot's data sheet).
  # The default is 1 (very conservative)
  max_speed_deviation: 400.0

  # Condition time [s]
  condition_time: 500.0

  # Duration for ramp in [s].
  ramp_in_time: 0.10

  # Position correction gain of the EGM controller (should be 0 for velocity controller).
  pos_corr_gain: 0.0
